### PR TITLE
fix(ci): Remove automation to initialize Docker on Mac

### PR DESCRIPTION
**Pull request by armenzg**
Docker has been doing all that is possible to prevent installing Docker Desktop
without user interaction, thus, we can't run `sentry devservices up`.
See https://github.com/docker/for-mac/issues/2359#issuecomment-910532550 for details.

This PR removes all code that at times had managed to automate the process on Mac.

We will rely on Ubuntu to test the `make bootstrap` path.

We also add caching for Brew.

### CODE REVIEW

**Comment by armenzg** [comment]
TODO: 

- [ ] Update bootstrap code
- [ ] Update dev documentation

File: .github/workflows/python-deps.yml [hunk 1]
```diff
@@ -26,17 +27,25 @@ jobs:
...
-          brew update && brew bundle -q
+          brew update -q && brew bundle -q
 
-      - name: Setup python
-        id: setup-python
+      - name: Setup Python
         uses: ./.github/actions/setup-python
-        with:
-          # XXX: We need to pass this python-deps-${{ matrix.os }}-py${{ matrix.python-version }}-${{ hashFiles('requirements-*.txt') }}
-          cache-files-hash: ${{ hashFiles('requirements-*.txt') }}
```
Comment (armenzg) [inline .github/workflows/python-deps.yml:39]: We should be fine using the default cache key.

**Comment by dashed** [comment]
⚰️

**Comment by armenzg** [comment]
> ⚰️

🤣

File: .github/workflows/development-environment.yml [hunk 2]
```diff
@@ -83,9 +75,4 @@ jobs:
           eval "$(pyenv init --path)"
           python -m venv .venv
           source .venv/bin/activate
-          make bootstrap
-
-      - name: Test direnv
```
Comment (armenzg) [inline .github/workflows/development-environment.yml:88]: Removing this as it does not seem to execute. I wonder if it ever did.

**Review by billyvg — APPROVED** [review]
:(

**Pull request merged**